azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -0,0 +1,979 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/base64"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+// TODO: confirm locking as appropriate
+
+func resourceLinuxVirtualMachine() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLinuxVirtualMachineCreate,
+		Read:   resourceLinuxVirtualMachineRead,
+		Update: resourceLinuxVirtualMachineUpdate,
+		Delete: resourceLinuxVirtualMachineDelete,
+		Importer: azSchema.ValidateResourceIDPriorToImportThen(func(id string) error {
+			_, err := ParseVirtualMachineID(id)
+			return err
+		}, importVirtualMachine(compute.Linux, "azurerm_linux_virtual_machine")),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(45 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(45 * time.Minute),
+			Delete: schema.DefaultTimeout(45 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: ValidateLinuxName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			// Required
+			"admin_username": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"network_interface_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: networkValidate.NetworkInterfaceID,
+				},
+			},
+
+			"os_disk": virtualMachineOSDiskSchema(),
+
+			"size": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			// Optional
+			"additional_capabilities": virtualMachineAdditionalCapabilitiesSchema(),
+
+			"admin_password": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Sensitive:        true,
+				DiffSuppressFunc: adminPasswordDiffSuppressFunc,
+			},
+
+			"admin_ssh_key": SSHKeysSchema(true),
+
+			"allow_extension_operations": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+
+			"availability_set_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.AvailabilitySetID,
+				// the Compute/VM API is broken and returns the Resource Group name in UPPERCASE :shrug:
+				DiffSuppressFunc: suppress.CaseDifference,
+				// TODO: raise a GH issue for the broken API
```

Comment:
intentionally not yet since I hadn't had time to put a detailed bug report together - was going to do this once the Beta's out and we can retroactively add this in